types/math/dec_test.go

```diff
@@ -89,28 +137,437 @@ func TestDec(t *testing.T) {
 	require.False(t, minusOne.IsZero())
 	require.False(t, minusOne.IsPositive())
 	require.True(t, minusOne.IsNegative())
+}
+
+// TODO: Think a bit more about the probability distribution of Dec
+var genDec *rapid.Generator = rapid.Custom(func(t *rapid.T) Dec {
+	f := rapid.Float64().Draw(t, "f").(float64)
+	dec, err := NewDecFromString(fmt.Sprintf("%g", f))
+	if err != nil {
+		t.Fatalf("Failed while generating Dec from float: %g", f)
+	}
+	return dec
+})
+
+// A Dec value and the float used to create it
+type floatAndDec struct {
+	float float64
+	dec   Dec
+}
+
+// Generate a Dec value along with the float used to create it
+var genFloatAndDec *rapid.Generator = rapid.Custom(func(t *rapid.T) floatAndDec {
+	f := rapid.Float64().Draw(t, "f").(float64)
+	dec, err := NewDecFromString(fmt.Sprintf("%g", f))
+	if err != nil {
+		t.Fatalf("Failed while generating Dec from float: %g", f)
```

Comment:
question: is there a reason why we don't do `require.NoError(t, err, "...")` here?

Reply:
i think this was for debugging purposes - I think we can safely remove it now